File: editorial/chap5-7/prob5-7-3.cpp
#include <iostream>
#include <algorithm>
using namespace std;
 
long long N, A[200009];
long long Answer = 0;
 
int main() {
	// 入力
	cin >> N;
	for (int i = 1; i <= N; i++) cin >> A[i];
	
	// ソート（コード 5.7.1 から追加した唯一の部分）
	sort(A + 1, A + N + 1);
	
	// 答えを求める → 答えの出力
	for (int i = 1; i <= N; i++) Answer += A[i] * (-N + 2LL * i - 1LL);
	cout << Answer << endl;
	return 0;
}

File: editorial/chap5-7/prob5-7-3.py
# 入力
N = int(input())
A = list(map(int, input().split()))

# 配列 A 全体をソートする
A.sort()

# 答えを求める
Answer = 0
for i in range(1, N + 1):
	Answer += A[i - 1] * (-N + 2 * i - 1)

# 出力
print(Answer)